services/wallet/service_test.go

```diff
@@ -19,8 +19,108 @@ import (
 	errorutils "github.com/brave-intl/bat-go/libs/errors"
 	"github.com/brave-intl/bat-go/libs/handlers"
 	"github.com/brave-intl/bat-go/services/wallet/model"
+	"github.com/brave-intl/bat-go/services/wallet/xslack"
 )
 
+func TestService_LinkSolanaAddress(t *testing.T) {
+	type tcGiven struct {
+		pid             uuid.UUID
+		req             linkSolanaAddrRequest
+		solAddrsChecker *mockAddrsChecker
+		compBotCl       compBotClient
+	}
+
+	type tcExpected struct {
+		err error
+	}
+
+	type testCase struct {
+		name  string
+		given tcGiven
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name: "error_address_not_allowed",
+			given: tcGiven{
+				solAddrsChecker: &mockAddrsChecker{
+					fnIsAllowed: func(ctx context.Context, addrs string) error {
+						return model.Error("error_address_not_allowed")
+					},
+				},
+			},
+			exp: tcExpected{
+				err: model.Error("error_address_not_allowed"),
+			},
+		},
+
+		{
+			name: "send_sol_sanctioned_message_error",
+			given: tcGiven{
+				solAddrsChecker: &mockAddrsChecker{
+					fnIsAllowed: func(ctx context.Context, addrs string) error {
+						return model.ErrSolAddrsNotAllowed
+					},
+				},
+				compBotCl: &xslack.MockClient{
+					FnSendMessage: func(ctx context.Context, msg *xslack.Message) error {
+						if msg == nil {
+							return model.Error("message_should_not_be_nil")
+						}
+
+						return model.Error("send_message_error")
+					},
+				},
+			},
+			exp: tcExpected{
+				err: model.Error("send_message_error"),
+			},
+		},
+
+		{
+			name: "send_sol_sanctioned_message_success",
+			given: tcGiven{
+				solAddrsChecker: &mockAddrsChecker{
+					fnIsAllowed: func(ctx context.Context, addrs string) error {
+						return model.ErrSolAddrsNotAllowed
+					},
+				},
+				compBotCl: &xslack.MockClient{
+					FnSendMessage: func(ctx context.Context, msg *xslack.Message) error {
+						if msg == nil {
+							return model.Error("message_should_not_be_nil")
+						}
+
+						return nil
+					},
+				},
+			},
+			exp: tcExpected{
+				err: model.ErrSolAddrsNotAllowed,
+			},
+		},
+	}
+
+	IsCheckerEnabled = true
```

Comment:
note: will remove this feat flag in followup PR tracking issue [here](https://github.com/brave-intl/bat-go/issues/2924).